### PR TITLE
OnnxTransform: Fix 3 bugbash bugs

### LINKED ISSUE #1050: OnnxTransform: Change namespace to Microsoft.ML.Transforms (currently Microsoft.ML.OnnxScoring)

### PR DESCRIPTION
**Pull request by shmoradims**
Fixes #1050 #1051 #1053

### CODE REVIEW

**Comment by Ivanidzo4ka** [comment]
> using Microsoft.ML.Runtime;

add header.

---
Refers to: src/Microsoft.ML.OnnxTransform/OnnxUtils.cs:1 in 6ffbc62. [](commit_id = 6ffbc62aceeab2f214bacbcac536314beca97a2e, deletion_comment = False)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs [hunk 1]
```diff
@@ -122,10 +128,16 @@ private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes =
...
+            if (modelInfo.InputsInfo.Length != 1)
+                throw env.Except($"OnnxTransform supports Onnx models with one input. The provided model has ${modelInfo.InputsInfo.Length} input(s).");
+            if (modelInfo.OutputsInfo.Length != 1)
+                throw env.Except($"OnnxTransform supports Onnx models with one output. The provided model has ${modelInfo.OutputsInfo.Length} output(s).");
+
             Input = args.InputColumn;
             Output = args.OutputColumn;
 
-            var outputNodeInfo = Model.GetOutputsInfo().Where(x => x.Name == args.OutputColumn).First();
+            var outputNodeInfo = Model.ModelInfo.OutputsInfo[0];
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: >ar outputNodeInfo = Model.ModelInfo.OutputsInfo[0]; [](start = 13, length = 51)

Why you made this change? Why you make this restriction on output column?

**Comment by shmoradims** [comment]
> using Microsoft.ML.Runtime;

Added

---
In reply to: [425256598](https://github.com/dotnet/machinelearning/pull/1080#issuecomment-425256598) [](ancestors = 425256598)

---
Refers to: src/Microsoft.ML.OnnxTransform/OnnxUtils.cs:1 in 6ffbc62. [](commit_id = 6ffbc62aceeab2f214bacbcac536314beca97a2e, deletion_comment = False)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs (hunk 1 above)
Comment (shmoradims) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: I removed the restriction. Now the onnx_model input/output node names don't matter, b/c there's only 1 input and 1 output. We just tie them to whatever IDV column is specified as input/output, w/o forcing the names to match. 

---
In reply to: [221091553](https://github.com/dotnet/machinelearning/pull/1080#discussion_r221091553) [](ancestors = 221091553)

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: How many models out there have only one input and one output? By imposing this restriction you denied any model which not suite this criteria.
What if user want's output from hidden layer in DNN, or some layer which is not last one in execution graph?

If problem with output node's names, why can you accept two strings for input and output, one as IDV column name, and other as ONNX name?

---
In reply to: [221109171](https://github.com/dotnet/machinelearning/pull/1080#discussion_r221109171) [](ancestors = 221109171,221091553)

Comment (jignparm) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: In Tensorflow, there are many examples of models with multiple inputs and outputs. But fo onnx , all the models in the zoo (see [https://github.com/onnx/models](https://github.com/onnx/models)) have exactly 1 in/output. . We can probably add support for multi-input/output support as part of ML.net v0.7, to make it future proof, but currently  cover 100% of all existing pretrained-onnx 1.2 models.

The change above gives more flexibility on how the user wants to name the IDV output column (i.e. doesn't have to be same name as model)

---
In reply to: [221112683](https://github.com/dotnet/machinelearning/pull/1080#discussion_r221112683) [](ancestors = 221112683,221109171,221091553)

Comment (shauheen) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: @wschin please review this.

**Comment by shauheen** [comment]
@shmoradims please use the correct notation when associating with multiple issues.

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs (hunk 1 above)
Comment (TomFinley) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: Sorry, do we disagree with @Ivanidzo4ka's proposal? It seems quite sensible to me. You have an output column name in IDV land, but what it corresponds to in tensorflow land is incidental.

As it happens, Ivan's design also makes it possible to make a well written pigstention, which currently is quite difficult under the current rather inflexible design.

**Comment by shmoradims** [comment]
sure

---
In reply to: [425318353](https://github.com/dotnet/machinelearning/pull/1080#issuecomment-425318353) [](ancestors = 425318353)

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs (hunk 1 above)
Comment (shmoradims) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: We resolved this after the scrum today with Ivan, Pete, Jignesh, Yael.

Conclusion: No change is needed.

Summary: B/c sonoma API only allows access to inputs and outputs (not intermediate layers), and all the onnx models in model zoo have just 1 input and 1 output, we can keep OnnxTransform API as is.

---
In reply to: [221142181](https://github.com/dotnet/machinelearning/pull/1080#discussion_r221142181) [](ancestors = 221142181)

**Pull request merged**